src/inference/src/dev/iremote_tensor.cpp

```diff
@@ -0,0 +1,358 @@
+#include "ie_ngraph_utils.hpp"
+#include "ie_remote_blob.hpp"
+#include "openvino/runtime/properties.hpp"
+
+namespace ov {
+
+IRemoteTensor::~IRemoteTensor() = default;
+
+/**
+ * @brief Tensor what contains InferenceEngine::Blob inside
+ * Blob owns the memory
+ */
+class BlobTensor : public ITensor {
+    mutable element::Type m_type;
+    mutable Shape m_shape;
+    mutable Strides m_strides;
+
+public:
+    std::shared_ptr<ie::Blob> blob;
+
+    BlobTensor(const InferenceEngine::Blob::Ptr& blob) : blob{blob} {
+        auto remote_impl = dynamic_cast<InferenceEngine::RemoteBlob*>(blob.get());
+        OPENVINO_ASSERT(!remote_impl);
+        OPENVINO_ASSERT(blob);
+        m_shape = blob->getTensorDesc().getBlockingDesc().getBlockDims();
+    }
+
+    const element::Type& get_element_type() const override {
+        m_type = InferenceEngine::details::convertPrecision(blob->getTensorDesc().getPrecision());
+        return m_type;
+    }
+
+    void set_shape(ov::Shape shape) override {
+        blob->setShape({shape.begin(), shape.end()});
+    }
+
+    const Shape& get_shape() const override {
+        m_shape = blob->getTensorDesc().getBlockingDesc().getBlockDims();
+        return m_shape;
+    }
+
+    const Strides& get_strides() const override {
+        OPENVINO_ASSERT(get_element_type().bitwidth() >= 8,
+                        "Could not get strides for types with bitwidths less then 8 bit. Tensor type: ",
+                        get_element_type());
+        const auto& element_strides = blob->getTensorDesc().getBlockingDesc().getStrides();
+        const size_t elem_size = get_element_type().size();
+        m_strides.clear();
+        m_strides.resize(element_strides.size());
+        std::transform(element_strides.begin(), element_strides.end(), m_strides.begin(), [&elem_size](size_t stride) {
+            return stride * elem_size;
+        });
+        return m_strides;
+    }
+
+    size_t get_size() const override {
+        return blob->size();
+    }
+
+    size_t get_byte_size() const override {
+        return blob->byteSize();
+    }
+
+    void* data(const element::Type& element_type) const override {
+        OPENVINO_ASSERT(blob != nullptr, "Tensor was not initialized.");
+#define TYPE_CHECK(TYPE) (dynamic_cast<const ie::TBlob<TYPE>*>(blob.get()) != nullptr)
+        auto host_accesable_implementation = TYPE_CHECK(bool) || TYPE_CHECK(int8_t) || TYPE_CHECK(uint8_t) ||
+                                             TYPE_CHECK(int16_t) || TYPE_CHECK(uint16_t) || TYPE_CHECK(int32_t) ||
+                                             TYPE_CHECK(uint32_t) || TYPE_CHECK(int64_t) || TYPE_CHECK(uint64_t) ||
+                                             TYPE_CHECK(float) || TYPE_CHECK(double);
+#undef TYPE_CHECK
+        OPENVINO_ASSERT(host_accesable_implementation,
+                        "Tensor implementation type dose not contains host accessable data");
+        if (element_type != element::undefined) {
```

Comment:
maybe we need also check element::dynamic here?
BTW, do we have tests that Tensors cannot be created with such strange types?